src/Units/UnitModelScheme.java

```diff
@@ -28,11 +28,20 @@ public String toString()
   public abstract String getIconicUnitName();
   
   // Fetches a shopping list and a full unit list for a generic commander.
-  public abstract GameReadyModels getGameReadyModels();
+  private GameReadyModels grms = null;
+  public final GameReadyModels getGameReadyModels()
+  {
+    if(grms != null)
+      return grms;
+    grms = buildGameReadyModels();
+    return grms;
+  }
+  public abstract GameReadyModels buildGameReadyModels();
```

Comment:
Why create this extra layer of indirection?

Reply:
It was an easy way to make everyone share the same `UnitModel`s without forcing every `UnitModelScheme` to do the caching independently or figure out somewhere else to put it?
The UMS instance will sit in the rules object for the lifetime of the game, so this seems like a reasonable solution to me.